prog-o-meter.py

```diff
@@ -144,13 +166,115 @@ def add_day(self):
+        """
+        Opens a Tkinter window for the user to enter the number of days they want to add to the prog-o-meter.
+        Window contains a submit button, for user to click when they have entered the number of days.
+        """
+        # Attributes
+        self.root = Tk.Tk()
+        self.days = _days
+        self.goal = goal
+        self.filename = filename
+        self.username = username
+        # Tkinter instantiation
+        self.canvas_layout()
+        self.submit_button()
+        self.root.mainloop()
+        
+    def canvas_layout(self):
+        """Display a Tkinter canvas.
+        
+        Creates a 300x50 px canvas with a greeting text and an entry widget (input field for text).
+        
+        Attributes:
+            CANVAS_WIDTH: The width of the canvas is hardcoded to 300 px.
+            CANVAS_HEIGHT: The height of the canvas is hardcoded to 50 px.
+            canvas: The Tkinter canvas widget.
+            text_entry: A Tkinter text-entry widget (input field for text)
+        """
+        self.CANVAS_WIDTH = 300
+        self.CANVAS_HEIGHT = 50
+        self.canvas = Tk.Canvas(self.root, width = self.CANVAS_WIDTH, height = self.CANVAS_HEIGHT)
+        self.canvas.pack()
+        self.text_entry = Tk.Entry(self.root)
+        self.text_entry.pack()
+        self.text_entry.focus_force()
+        self.canvas.create_text(self.CANVAS_WIDTH/2, 20, text = "Enter the days you want to progress")
+        
+    def submit_button(self):
+        """Display the submit button on the canvas.
+        
+        Displays a submit button, for user to click when they have entered number of days they want to add to their progress.
+        When button is clicked, it stores the input name as username, and then closes the Tkinter window.
+        
+        Attributes:
+            submit_button: Button with the text "Submit", which calls the function updateDays_and_close
+        """
+        self.submit_button = Tk.Button(self.root, text = "Submit", command = lambda: self.updateDays_and_close(self.days, self.filename, self.username))
+        self.submit_button.pack()
+        self.root.bind('<Return>', lambda: self.updateDays_and_close(self.days, self.filename, self.username))
+        
+    def updateDays_and_close(self, days, filename, username):
+        """Fill out multiple rectangles in prog-o-meter with blue color, to represent multiple days completed.
+        
+           Callback function to submit_button. Fills out multiple rectangles (from most left-ward white rectangle) with color blue, to represent multiple days completed.
+        """
+        update_day_number = int(self.text_entry.get())
+        days = days + update_day_number
+        self.root.destroy()
+        if days <= self.goal:
+            ProgressGUI(days, filename, username)
+            update_days_file(filename, days)
+        else:
+            print("Enter number between 0 and " + str(self.goal))

```

Comment:
Its awesome that you've added this to handle cases where the user don't act like we expect them to, but since the user doesn't have to use the Command line when running our program, printing to console is not that useful (most of them might not see it). So we should find a way to print this on a TKinter canvas, or not print is at all I think :)